app/views/matches/_client_files.haml

```diff
@@ -12,8 +12,10 @@
         - if required
           *
         - if file.present?
-          - label = "#{tag} (#{file.effective_date})"
-          = link_to tag, Config.get(:warehouse_url) + "/clients/#{client.remote_id}/files/#{file.id}", target: '_blank'
+          - if file.effective_date.present?
+            - date = " (#{date_format(file.effective_date)})"
+          - label = "#{tag}#{date}"
+          = link_to label, Config.get(:warehouse_url) + "/clients/#{client.remote_id}/files/#{file.id}", target: '_blank'
```

Comment:
Let's push the link calculation into the file class, this looks really brittle, but if we have it in one place, it'll be easier to maintain.